services/api/src/index.ts

```diff
@@ -1506,40 +1628,35 @@ server.after(() => {
         url: `/api/${apiVersion}/user/delete`,
         schema: {},
         handler: async (request, reply) => {
-            const { didWrite, encodedUcan } = await verifyUCAN(db, request, {
-                expectedDeviceStatus: undefined,
-            });
-
-            const status = await authUtilService.isLoggedIn(db, didWrite);
-            if (!status.isLoggedIn) {
-                throw server.httpErrors.unauthorized("Device is not logged in");
-            } else {
-                await deleteUserAccount({
-                    proof: encodedUcan,
-                    db: db,
-                    didWrite: didWrite,
-                    userId: status.userId,
-                    baseImageServiceUrl: config.IMAGES_SERVICE_BASE_URL,
+            const { didWrite, encodedUcan, deviceStatus } =
+                await verifyUcanAndKnownDeviceStatus(db, request, {
```

Comment:
This function is used to return a deviceStatus that will contain the `userId`, since the device is expected to be "known".